### PR TITLE
Added operating mode config option & behavior.

### PR DESCRIPTION
**Pull request by bsneed**
- Added operatingMode confi option & logic
- Changed unit-test settings setup to respect operating mode.
- Made some startup plugins not auto-load when in server mode.
- Fixed logging output to be correct.
- Updated testPurgeStorage to reflect new flush behavior.
- Added test for server operation mode.

### CODE REVIEW

File: Sources/Segment/Configuration.swift [hunk 1]
```diff
@@ -10,6 +10,15 @@ import Foundation
 import FoundationNetworking
 #endif
 
+// MARK: - Operating Mode
+/// Specifies the operating mode/context
+public enum OperatingMode {
+    /// Running as an executable on a server
```
Comment (didiergarcia) [inline Sources/Segment/Configuration.swift]: Callout to project README.md to list differences.

**Review by didiergarcia — APPROVED** [review]

**Review by alanjcharles — APPROVED** [review]

File: Sources/Segment/Configuration.swift [hunk 2]
```diff
@@ -10,6 +10,15 @@ import Foundation
...
+    case server
+    /// Running as a long-lived application
```
Comment (oscb) [inline Sources/Segment/Configuration.swift]: One could argue a server is longer-lived than an app! stateless might be a better word.

The other thing I'm thinking of is, technically you might want to do async in a server too. At least in other languages I have used for server side.

Comment (bsneed) [reply Sources/Segment/Configuration.swift]: I definitely see your point.  Would better enum values be sync/async? cc @didiergarcia

Comment (oscb) [reply Sources/Segment/Configuration.swift]: I think async/sync make it clearer what to expect when you switch!

**Pull request merged**